imkar/analytical/analytical.py

```diff
@@ -0,0 +1,169 @@
+    h = height_to_length*length #height of the profile
+    #vector of corresponding wavelength in air
+    lambda_air = c/frequency_vector[:] 
+    k = 2*np.pi/lambda_air #vector of wavenumbers
+
+    eps = 1E-3 # stop criterion for infinite sum over u
+    n_max = 50 # truncation parameter for numbers of outgoing waves
+    n_max2 = 2*n_max+1 #counter from -N to N including zero
+
+    x_u = np.zeros((1, n_max),dtype=complex)
+    n = np.arange(-n_max,n_max+1)[:, np.newaxis]
+    previous_frequency = 0
+
+    n_frequency = frequency_vector.size # number of computed frequencies
+    n_phi = phi_vector.size # number of computed incident angles
+    s = np.zeros((n_frequency, n_phi))
+
+    #calculation
+    for iFrequencies in range(1, (n_frequency+1)):
+        for iPhi in range(1, (n_phi+1)):
+            phi_0 = phi_vector[iPhi-1]
+            alpha_n = np.cos(phi_0) + n*lambda_air[iFrequencies-1]/length
+            beta_n = np.conj(-(1j*np.emath.sqrt(alpha_n**2-1)))            
+
+            #helping variables
+            k_frequency = k[iFrequencies-1]
+            k_alpha_n = (k_frequency*alpha_n[:]).T
+            jk_alpha_n_over_width = 1j*k_alpha_n/width
+
+            y_nr = np.zeros((n_max2, n_max2),dtype=complex)
+
+            for iN in range(1, (n_max2+1)):
+                u_max = np.zeros((1, n_max2),dtype=int)
+                u_max_element = 0
+                for iR in range(1, (n_max2+1)):
+                    u_max[0,iR-1] = 0
+                    prevTempU = 0
+                    err = 1
+                    while err>eps:
+                        u_max[0,iR-1] = u_max[0,iR-1] + n_max2-1
+
+                        if iR==1:
+                            U_u_n = np.zeros((u_max.item((iR-1))+1,n_max2),\
+                                dtype=complex)
+                        elif u_max[0,iR-1]>u_max_element:
+                            U_u_n = np.zeros((u_max.item((iR-1))+1, n_max2),\
+                                dtype=complex) 
+                        
+                        if ((iR==1) or (u_max[0,iR-1]>u_max_element) or \
+                            (iFrequencies>previous_frequency)):
+                            iU = np.arange(0., round(u_max.item((iR-1)))+1,\
+                                dtype=int)[:, np.newaxis]                           
+                            uVals = np.tile(iU, (1, n_max2))
+                            k_x_u = iU*np.pi/width
+                            x_u = np.emath.sqrt(k_frequency**2 - k_x_u**2)
+
+                            CaseA = np.less(abs((k_alpha_n-k_x_u)), 1E-3)
+                            CaseA = CaseA.astype(int)
+                            CaseB = np.less(abs((k_alpha_n+k_x_u)), 1E-3)
+                            CaseB = CaseB.astype(int)
+                            CaseC = np.ones(np.shape(CaseA))-(CaseA+CaseB)
+                            CaseC = CaseC.astype(int) 
+                            if(np.sum(CaseA)>0):                           
```

Comment:
```suggestion
                            if np.sum(CaseA) > 0:
```
brackets are not needed in python

Reply:
Changed.